python/adbc_driver_manager/adbc_driver_manager/dbapi.py

```diff
@@ -926,6 +926,19 @@ def fetch_df(self) -> "pandas.DataFrame":
             )
         return self._results.fetch_df()
 
+    def fetch_query_schema(self, operation: Union[bytes, str], parameters=None) -> pyarrow.Schema:
+        """
+        Fetch the output schema for the given query.
+
+        Notes
+        ----
+        This is an extension and not part of the DBAPI standard.
+        """
+        self._prepare_execute(operation, parameters)
+        handle, _ = self._stmt.execute_query()
+        rbr = pyarrow.RecordBatchReader._import_from_c(handle.address)
```

Comment:
This will still execute the query. If that's what you want, then just execute the query explicitly instead of this, which is just a method that lies to the caller.